graylog2-server/src/main/java/org/graylog2/search/SearchQueryParser.java

```diff
@@ -231,15 +233,18 @@ private DateTime parseDate(String value) {
     FieldValue createFieldValue(SearchQueryField field, String quotedStringValue, boolean negate) {
         // Make sure there are no quotes in the value (e.g. `"foo"' --> `foo')
         final String value = quotedStringValue.replaceAll(QUOTE_REPLACE_REGEX, "");
-        final Pair<String, SearchQueryOperator> pair = extractOperator(value, DEFAULT_OPERATOR);
+        final SearchQueryField.Type fieldType = field.getFieldType();
+        final Pair<String, SearchQueryOperator> pair = extractOperator(value, fieldType == STRING ? DEFAULT_STRING_OPERATOR : DEFAULT_OPERATOR);
 
-        switch (field.getFieldType()) {
+        switch (fieldType) {
             case DATE:
                 return new FieldValue(parseDate(pair.getLeft()), pair.getRight(), negate);
             case STRING:
                 return new FieldValue(pair.getLeft(), pair.getRight(), negate);
+            case INT:
+                return new FieldValue(Integer.parseInt(pair.getLeft()), pair.getRight(), negate);
```

Comment:
Can we also use a `long` instead of an `int` here? This would make it work with very large values as well.

If MongoDB works with both, we could just parse the value into a `long` and rename the data type to `NUMBER` to make it more generic.

Reply:
So far I only needed an `int` and I tried to only implement that. I can look into it, though.

Anyway, I think `NUMBER` is not really good even with `long` support, because then I would also expect the method to work with `double`, `float`, and maybe other number types in Java. Also makes parsing the number a bit more of a challenge 😄

Reply:
That didn't work when parsing a number as `long` and then trying to serialize it to a POJO that has an `int` field. Mongojack tries to serialize the `long` value as an `int` and it fails with:
`java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer`